engines/libaio.c

```diff
@@ -366,27 +444,26 @@ static int fio_libaio_init(struct thread_data *td)
 	ld->io_us = calloc(ld->entries, sizeof(struct io_u *));
 
 	td->io_ops_data = ld;
+	/*
+	 * Initialize prio_percent mechanisms if relevant
+	 */
+	if (!(fio_option_is_set(to, ioprio) ||
+			fio_option_is_set(to, ioprio_class) ||
+			(o->prio_percent == 0))) {
+		ioengine.prio_prep = fio_libaio_prio_prep;
+		ld->prio_seed = FIO_RANDSEED * td->thread_number
```

Comment:
Shouldn't we be using fio's built-in random number generation for this - (see the functions in `lib/rand.h`)? Then `FIO_RANDSEED` wouldn't need duplicating...

Reply:
It isn't clear to me how to use fio's built-in random number generation to seed the random number generator. Wouldn't I need to set up another random state to use any of the random number generators, which would in turn need its own seed? I'm not particularly attached to using the FIO_RANDSEED value, but I was trying to follow the way random seeds generally seem to be chosen. Should I just pick an arbitrary constant value?